tango_simlib/sim_xmi_parser.py

```diff
@@ -224,6 +240,49 @@ def parse(self, sim_xmi_file):
                     class_description_data, class_description_data.tag)
                 self.class_properties.append(class_property_info)
 
+    def extract_device_class_descr(self, description_data):
+        """Extract Tango device class description data from the xmi tree element.
+
+        Parameters
+        ----------
+        description_data: xml.etree.ElementTree.Element
+            XMI tree element with class description data, where
+            expected element tag(s) are (i.e. description_data.tag)
+            ['inheritances(s)', 'identification'] and
+            description_data.attrib contains
+            {
+                "description": "",
+                "title": "",
+                "sourcePath": "",
+                "language": "",
+                "filestogenerate": "",
+                "license": "",
+                "copyright": "",
+                "hasMandatoryProperty": "",
+                "hasConcreteProperty": "",
+                "hasAbstractCommand": "",
+                "hasAbstractAttribute" : ""
+            }
+
+        Returns
+        -------
+        class_data: dict
+            Dictionary of all the class device data required for additional device
+            information.
+
+        """
+        #class_data = description_data.attrib   # This contains the additional
+                                                # information about the Tango device
+                                                # class, however it is not useful for
+                                                # the current problem.
+        class_data = {}
+        super_classes = description_data.findall('inheritances')
+        for super_class, super_class_num in zip(
+                super_classes, range(len(super_classes))):
```

Comment:
This is quite a hard way of getting a counter.  You could also use the handy enumerate method:
```
for num, super_class in enumerate(super_classes):
```